git-link.el

```diff
@@ -666,6 +666,54 @@ return (FILENAME . REVISION) otherwise nil."
     (git-link--remote)))
 
 ;;;###autoload
+(defun git-link-url (file-or-buffer remote &optional start end use-branch-p)
+  "Create a URL for the FILE-OR-BUFFER's location at REMOTE.
+
+When given START and END, include the current line number or
+active region.
+
+When given USE-BRANCH-P, force the use of the branch name in the
+resultant url."
+  (let* ((filename (git-link--relative-filename
+                    (if (bufferp file-or-buffer)
+                        (buffer-file-name file-or-buffer)
+                      file-or-buffer)))
+         (remote-url (git-link--remote-url remote))
```

Comment:
We must check for a `nil` `remote-url` first, like we do here: https://github.com/sshaw/git-link/pull/69/files#diff-a86ea92b0e180064b421a637b78c7d5ae152de93c60a82136b77a6a72c90b35cL692
Otherwise we get errors when the remote does not exist.

Reply:
A case where the remote does not exist would be trying to link from a buffer that is not part of a git repo.